sdk/src/test/java/com/horizen/proposition/CustomPropositionTest.java

```diff
@@ -0,0 +1,47 @@
+package com.horizen.proposition;
+
+import com.horizen.customtypes.CustomPrivateKey;
+import com.horizen.secret.PrivateKey25519;
+import com.horizen.secret.Secret;
+import com.horizen.utils.Ed25519;
+import com.horizen.utils.Pair;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import static org.junit.Assert.*;
+
+public class CustomPropositionTest {
```

Comment:
I think, that we should add custom proposition that can be opened by k of n secrets. Like what we have in Lambo Registry for sell order, that can be opened or by the owner or by the buyer.
Then add the test similar to this one.